test/suites/clustering.sh

```diff
@@ -4232,3 +4232,82 @@ test_clustering_trust_add() {
   kill_lxd "${LXD_ONE_DIR}"
   kill_lxd "${LXD_TWO_DIR}"
 }
+
+test_clustering_link() {
+  local LXD_DIR
+
+  LXD_ONE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_ONE_DIR}"
+  spawn_lxd "${LXD_ONE_DIR}" false
+
+  LXD_TWO_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  chmod +x "${LXD_TWO_DIR}"
+  spawn_lxd "${LXD_TWO_DIR}" false
+
+  # shellcheck disable=SC2034,SC2030
+  LXD_DIR=${LXD_ONE_DIR}
+
+  # Get the address of LXD_ONE.
+  LXD_ONE_ADDR="$(lxc config get core.https_address)"
+
+  # Enable clustering on LXD_ONE.
+  lxc cluster enable node1
+  [ "$(lxc cluster list | grep -cF 'node1')" = 1 ]
+
+  # Get a cluster link trust token from LXD_ONE.
+  lxd_one_trust_token="$(lxc cluster link add lxd_two --quiet)"
+  echo "$lxd_one_trust_token" > "${TEST_DIR}/lxd_one_trust_token"
+
+  # The cluster link identity on LXD_ONE should be pending.
+  [ "$(lxc auth identity list --format csv | grep -cF 'Cluster link certificate (pending)')" = 1 ]
+
+  LXD_DIR=${LXD_TWO_DIR}
+
+  # Get the address of LXD_TWO.
+  LXD_TWO_ADDR="$(lxc config get core.https_address)"
+
+  # Enable clustering on LXD_TWO.
+  lxc cluster enable node1
+  [ "$(lxc cluster list | grep -cF 'node1')" = 1 ]
+
+  # Add cluster link on LXD_TWO using the token from LXD_ONE.
+  # shellcheck disable=SC2002
+  lxc cluster link add lxd_one --token "$(cat "${TEST_DIR}/lxd_one_trust_token" | tr -d '\n')"
```

Comment:
```suggestion
  lxc cluster link add lxd_one --token "$(tr -d '\n' < "${TEST_DIR}/lxd_one_trust_token")"
```
But I'm not sure that's the original format and what's the desired one. Why does it need massaging in the first place?

Reply:
I thought there was an extra newline being added to the file but it looks like there isn't. `cat` will work fine :)